src/src/PluginStructs/P062_data_struct.cpp

```diff
@@ -2,24 +2,62 @@
 
 
 #ifdef USES_P062
+#include "../Helpers/ESPEasy_Storage.h"
 
-P062_data_struct::P062_data_struct(uint8_t i2c_addr, bool scancode) : use_scancode(scancode) {
-  keypad.begin(i2c_addr);
+P062_data_struct::P062_data_struct(uint8_t i2c_addr, bool scancode, bool keepCalibrationData) 
+    : i2c_addr(i2c_addr), use_scancode(scancode), keepCalibrationData(keepCalibrationData) {
```

Comment:
Better use slightly different names for the constructor parameters and the members you try to initialize.

I must admit there isn't lot of consistency in our code, so maybe we can agree (now) on some convention?
For examples, use a `_` prefix on member variables, like `_use_scancode` as member.

Reply:
That's a good convention, should have thought of that myself but I added it in the same style, though still different. Will fix.

Reply:
I was also thinking of moving all arguments to the `init()` function, as none of them are actually used anymore in the constructor. When having the instantiation of the keypad in the constructor, my ESP(32) just hangs as I don't have the hardware attached (though the driver should actually handle that, but it seems it doesn't).

Reply:
> When having the instantiation of the keypad in the constructor, my ESP(32) just hangs as I don't have the hardware attached (though the driver should actually handle that, but it seems it doesn't).

Hmm that's absolutely not ideal as a user may play around, or hardware connections get loose.
That should be fixed before merge... if possible.

Reply:
Move the arguments to `init()`

> Hmm that's absolutely not ideal as a user may play around, or hardware connections get loose.
> That should be fixed before merge... if possible.

I agree, and I've taken a look at that code, but there isn't any obvious cause that _I_ can recognize.
Could you have a look at the Adafruit_MPR121 sources?